### ISSUE TITLE
no-loop-func: Doesn't warn with for-of loops

### ISSUE THREAD

**Issue by andersekdahl** [eslint/eslint#1859]
Version 0.14.1 of eslint with the following code:

```
'use strict';

for (let i of []) {
  let func = (x) => console.log(x);
  func(i);
}
```

Doesn't produce an any error with `no-loop-func` activated, whereas this code does:

```
'use strict';

for (let i in []) {
  let func = (x) => console.log(x);
  func(i);
}
```

Produces: `4:13  error  Don't make functions within a loop  no-loop-func`. Simply adding a check for `ForOfStatement` in the rule together with the other checks fixes it.

**Comment by nzakas** [comment]
Definitely a bug, thanks.

**Comment by lo1tuma** [comment]
Working on this.
